source/NVDAObjects/UIA/VisualStudio.py

```diff
@@ -0,0 +1,39 @@
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2020 NV Access Limited, Leonard de Ruijter
+
+"""
+Object overlay classes for Visual Studio components
+available in Visual Studio and SQL Server Management Studio.
+"""
+
+from . import UIA
+import speech
+import braille
+import api
+
+
+class IntelliSenseItem(UIA):
+
+	def _get_name(self):
+		return self.UIAElement.cachedAutomationID
+
+	def event_UIA_elementSelected(self):
+		speech.cancelSpeech()
```

Comment:
what speech is this trying to cancel? I assume the selection message?

Reply:
Yes. If we wouldn't cancel speech, al selection announcements would queue up when arrowing up/down.

The logic was actually borrowed from UIA.SuggestionListItem, but I'm pretty sure that controllerFor isn't used in SSMS. I will double check, though.

Reply:
Nope, doesn't work when inheriting from SuggestionListItem

Reply:
Please add a comment to explain that it is the item selection speech this intends to cancel, and why this must cancel it and replace it with a different announcement.

Reply:
```suggestion
		# Cancel speech to have speech announce the selection as soon as possible.
		# This is needed because L{reportFocus} does not cancel speech.
		# Therefore, if speech wouldn't be cancelled,
		# selection announcements would queue up when changing selection rapidly.
		speech.cancelSpeech()
```